compiler/noirc_evaluator/src/ssa/checks/check_for_underconstrained_values.rs

```diff
@@ -263,12 +276,11 @@ impl BrilligTaintedIds {
         }
 
         // Along with it, one of the argument descendants should be constrained
-        // (skipped if there were no arguments, or if an actual result and not a
-        // descendant has been constrained _alone_, e.g. against a constant)
+        // (skipped if there were no arguments, or if a result descendant
+        // has been constrained _alone_, e.g. against a constant)
         if !results_involved.is_empty()
             && (self.arguments.is_empty()
-                || (constrained_values.len() == 1
-                    && self.root_results.intersection(constrained_values).next().is_some())
+                || (constrained_values.len() == 1)
                 || self.arguments.intersection(constrained_values).next().is_some())
```

Comment:
Why is it no longer required for the arguments to contain the single constrained value?

Reply:
Ah, missed the `root_results`, I thought `arguments` were detached 🙈 

(Maybe a helper function for `this_set.intersection(that_set).next().is_some()` would be helpful, it appears 6 (now 5) times).